### PR TITLE
Issue#10 Processes POST requests and implements git cloning

### LINKED ISSUE #10: Clone the repository
Make the CI-server clone the repository.

### LINKED ISSUE #16: Retrieve branch name from request

### PR DESCRIPTION
**Pull request by asta12**
Adds a function which processes the request and retrieves the JSON in the POST request from git. It also adds git cloning to the CI-Server.
Closes #10 
Closes #16

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,13 +12,10 @@ plugins {
...
 
-ext {
-   javaMainClass = "ContinuousIntegrationServer"
-}
 
 application {
-    mainClassName = 'ContinuousIntegrationServer'
-}
+    mainClass.set("ContinuousIntegrationServer")
+
```
Comment (AnnaSkantz) [inline build.gradle:18]: I think "}" is missing on this row.

File: src/main/java/ContinuousIntegrationServer.java [hunk 2]
```diff
@@ -24,22 +37,91 @@ public void handle(String target,
...
+                System.out.println(branch);
+            }
+        }
+
+        cloneRepo(branch);
+        // how to call gradle and print output
+        String compileOutput = processCall("./CI-Server/gradlew compileJava");
+        System.out.println(compileOutput);
+        String testOutput = processCall("./CI-Server/gradlew test");
+        System.out.println(testOutput);
```
Comment (AnnaSkantz) [inline src/main/java/ContinuousIntegrationServer.java:56]: Should we really clone the repo even though the request is neither a "POST" request nor a "push" event? What do you think about moving the code on lines 51-56 into the if-statements above?

Comment (asta12) [reply src/main/java/ContinuousIntegrationServer.java:56]: I think that is a good idea. We  should only do it if something has been pushed.

File: build.gradle (hunk 1 above)
Comment (asta12) [reply build.gradle:18]: Yes it does. I accidentally removed it when I fixed merge conflicts.

**Comment by joakimAbdinurIusuf** [comment]
LGTM

**Pull request merged**